src/backend/core/observability.py

```
@@ -0,0 +1,153 @@
+            token = context.attach(ctx)
+            try:
+                yield span
+            finally:
+                span.end()
+                context.detach(token)
+
+        return _managed_span()
+
+    except ImportError:
+        return contextlib.nullcontext()
+    except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)